dashboard/test/controllers/api/v1/projects/public_gallery_controller_test.rb

```diff
@@ -1,149 +1,93 @@
-  published_applab_project = nil
   setup do
     published_applab_project = {
       storage_id: 22,
       id: 33,
-      published_at: '2017-03-03T00:00:00.000-08:00',
-      project_type: 'applab',
-      value: {
-        name: 'Charlies App',
-        thumbnailUrl: '/v3/files/charlies_thumbnail.png'
-      }.to_json,
-      name: 'Prince Charles',
-      birthday: 13.years.ago.to_datetime,
-      abuse_score: 0
+      name: "Applab1_to_be_featured",
+      thumbnailUrl: "/v3/files/thumbnail.png",
+      type: "applab",
+      publishedAt: "2023-02-14T02:14:51.000+00:00",
+      studentName: "A",
+      studentAgeRange: "18+",
+      isFeatured: true
     }
-    stub_projects = [published_applab_project]
-
-    # return 1 applab project for initial DB request, then an empty set
-    # for all subsequent db requests. Subsequent requests are only expected
-    # when 'all' project types are requested.
-    Projects.stubs(:table).returns(db_result(stub_projects)).then.returns(db_result([]))
+    all_featured_projects = {applab: [published_applab_project], gamelab: [], spritelab: [], playlab: [], artist: [], minecraft: [], events: [], k1: [], dance: [], poetry: [], library: []}
+    applab_featured_projects = {applab: [published_applab_project]}
+    ProjectsList.stubs(:fetch_featured_published_projects).returns(all_featured_projects)
+    ProjectsList.stubs(:fetch_featured_projects_by_type).returns(applab_featured_projects)
   end
 
   test_user_gets_response_for(
     :index,
     name: 'anonymous user can list applab projects',
-    params: -> {{project_type: 'applab', limit: 1}},
-  )
-
-  test_user_gets_response_for(
-    :index,
-    name: 'anonymous user can list gamelab projects',
-    params: -> {{project_type: 'gamelab', limit: 1}},
-  )
-
-  test_user_gets_response_for(
-    :index,
-    name: 'anonymous user can list playlab projects',
-    params: -> {{project_type: 'playlab', limit: 1}},
-  )
-
-  test_user_gets_response_for(
-    :index,
-    name: 'anonymous user can list artist projects',
-    params: -> {{project_type: 'artist', limit: 1}},
-  )
-
-  test_user_gets_response_for(
-    :index,
-    name: 'anonymous user can specify published_before',
-    params: -> {{project_type: 'applab', limit: 1, published_before: '2017-02-17T13:59:11.000-08:00'}},
+    params: -> {{project_type: 'applab'}},
   )
 
   test_user_gets_response_for(
     :index,
     name: 'anonymous user can list all published projects',
-    params: -> {{project_type: 'all', limit: 1}},
+    params: -> {{project_type: 'all'}},
   )
 
   test_user_gets_response_for(
     :index,
     name: 'bad request when project type is invalid',
-    params: -> {{project_type: 'bogus', limit: 1}},
-    response: :bad_request,
-  )
-
-  test_user_gets_response_for(
-    :index,
-    name: 'bad request when published_before is specified for project type all',
-    params: -> {{project_type: 'all', limit: 1, published_before: '2017-02-17T13:59:11.000-08:00'}},
```

Comment:
Now `featured_before` can be specified for project type all because in the future, there may be a 'View more' link for featured projects according to @samantha-code .